### PR TITLE
Add FreeImage 3.18.0

### PR DESCRIPTION
**Pull request by shuryanc**
FreeImage is an Open Source library project for developers to support popular graphics image formats. I will need this as dependecy for MegaSync.

### CODE REVIEW

File: components/library/FreeImage/Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+COMPONENT_ARCHIVE_URL=	http://downloads.sourceforge.net/freeimage/$(COMPONENT_ARCHIVE)
+COMPONENT_ARCHIVE_HASH=	sha256:f41379682f9ada94ea7b34fe86bf9ee00935a3147be41b6569c9605a53e438fd
+COMPONENT_LICENSE=	GPLv3
+COMPONENT_LICENSE_FILE=	license-gplv3.txt
+
+
+include $(WS_TOP)/make-rules/prep.mk
+include $(WS_TOP)/make-rules/justmake.mk
+include $(WS_TOP)/make-rules/ips.mk
+
```
Comment (AndWac) [inline components/library/FreeImage/Makefile:37]: Please also update to the new style here.

**Comment by shuryanc** [comment]
Rebased.

**Comment by AndWac** [comment]
This does not successfully publish here:

ginstall -c -m 644 Source/FreeImage.h /export/home/andreas/oi-userland/components/library/FreeImage/build/prototype/i386/usr/include
ginstall: cannot stat '/export/home/andreas/oi-userland/components/library/FreeImage/build/prototype/i386/usr/include/FreeImage.h': Permission denied
make[2]: *** [Makefile.gnu:75: install] Error 1

As a side note: Makefile, FreeImage.p5m, and libfreeimage.pc have executable permissions.

**Comment by shuryanc** [comment]
Fixed.

File: components/library/FreeImage/FreeImage.p5m [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=usr/include/FreeImage.h
+file path=usr/lib/$(MACH64)/libfreeimage-3.18.0.so
+file path=usr/lib/$(MACH64)/libfreeimage.a
+link path=usr/lib/$(MACH64)/libfreeimage.so target=libfreeimage.so.3
```
Comment (AndWac) [inline components/library/FreeImage/FreeImage.p5m:27]: We only deliver static libraries when no other options are available. Do you have a concrete need for static libraries? If not, then please either suppress their build or add appropriate drop transforms.

**Comment by shuryanc** [comment]
Removed static library.

**Pull request merged**